### PR TITLE
Start tmpnb containers in dev by default

### PR DESCRIPTION
**Pull request by jtyberg**
Update Makefile to start the tmpnb service by default.  Updated README too.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,8 +62,10 @@ run: | build run-kernel-gateway
...
 		-e HTTPS_KEY_FILE=$(HTTPS_KEY_FILE) \
 		-e HTTPS_CERT_FILE=$(HTTPS_CERT_FILE) \
-		-e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):8888 \
+		-e KERNEL_GATEWAY_URL=$(KERNEL_GATEWAY_URL) \
+		-e KERNEL_CLUSTER_URL=$(KERNEL_CLUSTER_URL) \
 		--link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
+		--link $(TMPNB_PROXY_CONTAINER_NAME):$(TMPNB_PROXY_CONTAINER_NAME) \
 		$(DASHBOARD_IMAGE_NAME) $(CMD)
 
 run-debug: CMD=start-debug
```
Comment (jhpedemonte) [inline Makefile]: We're passing both `GATEWAY` and `CLUSTER` URLs? How do we determine (within node app) when to connect to cluster and when to connect to single kernel gateway?

Comment (parente) [inline Makefile]: I was suggesting:
```
# Targets for running the nodejs app and kernel gateway in containers
_run: HTTP_PORT?=3000
_run: HTTPS_PORT?=3001
_run: | build run-kernel-gateway run-tmpnb
	@docker run -it --rm \
		--name $(DASHBOARD_CONTAINER_NAME) \
		-p $(HTTP_PORT):$(HTTP_PORT) \
		-p $(HTTPS_PORT):$(HTTPS_PORT) \
		-p 9711:8080 \
		-e PORT=$(HTTP_PORT) \
		-e HTTPS_PORT=$(HTTPS_PORT) \
		-e HTTPS_KEY_FILE=$(HTTPS_KEY_FILE) \
		-e HTTPS_CERT_FILE=$(HTTPS_CERT_FILE) \
                -e $(KERNEL_ENV_VAR) \
		--link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
		--link $(TMPNB_PROXY_CONTAINER_NAME):$(TMPNB_PROXY_CONTAINER_NAME) \
		$(DASHBOARD_IMAGE_NAME) $(CMD)

run: KERNEL_ENV_VAR='KERNEL_GATEWAY_URL=$(KERNEL_GATEWAY_URL)'
run: _run

run-tmpnb: KERNEL_ENV_VAR='KERNEL_CLUSTER_URL=$(KERNEL_CLUSTER_URL)'
run-tmpnb: _run
```

Comment (parente) [inline Makefile]: Or a pattern I like more lately (see https://github.com/jupyter-incubator/kernel_gateway/blob/master/Makefile):

```
define DOCKER
docker run -it --rm \
        --name $(DASHBOARD_CONTAINER_NAME) \
        -p $(HTTP_PORT):$(HTTP_PORT) \
        -p $(HTTPS_PORT):$(HTTPS_PORT) \
        -p 9711:8080 \
        -e PORT=$(HTTP_PORT) \
        -e HTTPS_PORT=$(HTTPS_PORT) \
        -e HTTPS_KEY_FILE=$(HTTPS_KEY_FILE) \
        -e HTTPS_CERT_FILE=$(HTTPS_CERT_FILE) \
        --link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
        --link $(TMPNB_PROXY_CONTAINER_NAME):$(TMPNB_PROXY_CONTAINER_NAME)
endef

run:
  $(DOCKER) -e KERNEL_GATEWAY_URL=$(KERNEL_GATEWAY_URL) $(DASHBOARD_IMAGE_NAME) $(CMD)

run-tmpnb:
  $(DOCKER) -e KERNEL_CLUSTER_URL=$(KERNEL_CLUSTER_URL) $(DASHBOARD_IMAGE_NAME) $(CMD)
```

Comment (jtyberg) [inline Makefile]: I'll change the make targets to start the node app with **either** the kernel gateway or tmpnb.

Regardless of the make targets, the code has to check for `CLUSTER_URL`, then `GATEWAY_URL`, then #fail.

**Comment by parente** [comment]
This is so dev on #11 can proceed cleanly.

**Comment by jtyberg** [comment]
OK. With a little guidance from @parente, I think I got it right this time.

**Pull request merged**